src/os/bluestore/BlueStore.cc

```diff
@@ -6028,21 +6028,24 @@ int BlueStore::_do_alloc_write(
 	csum_order = std::min(wctx->csum_order, ctz(l->length()));
       }
     }
-    while (final_length > 0) {
-      bluestore_pextent_t e;
-      uint32_t l;
-      uint64_t want = max_alloc_size ? MIN(final_length, max_alloc_size) : final_length;
-      int r = alloc->allocate(want, min_alloc_size, hint,
-			      &e.offset, &l);
-      assert(r == 0);
-      need -= l;
-      e.length = l;
+
+    int count = 0;
+    std::vector<AllocExtent> extents = 
```

Comment:
Resizing extent add possibility of realloc. I wanted to keep that away, max size of vector is simple calculation size / min alloc size.  I think assert catches miscalculation right not and any new call.